src/ophyd_async/panda/hdf_panda.py

```diff
@@ -0,0 +1,49 @@
+from __future__ import annotations
+
+from typing import Sequence
+
+from ophyd_async.core import (
+    DEFAULT_TIMEOUT,
+    DirectoryProvider,
+    SignalR,
+    StandardDetector,
+)
+from ophyd_async.epics.pvi import fill_pvi_entries, pre_initialize_blocks
+
+from .common_panda import CommonPandaBlocks
+from .panda_controller import PandaPcapController
+from .writers.hdf_writer import PandaHDFWriter
+
+
+class HDFPanda(CommonPandaBlocks, StandardDetector):
+    def __init__(
+        self,
+        prefix: str,
+        directory_provider: DirectoryProvider,
+        config_sigs: Sequence[SignalR] = (),
+        name: str = "",
+    ):
+        self._prefix = prefix
+        self.set_name(name)
+
+        pre_initialize_blocks(self, included_optional_fields=("data",))
```

Comment:
What is the purpose of `included_optional_fields`?

Reply:
Leftover from when we made `data` `Optional` in `CommonPandaBlocks`, we don't need this arg anymore.

The idea of the argument is that we won't want to initialize most optional fields at `__init__` time. If the Panda didn't send such a block then you'd have blocks without signals at connect time. This is the argument used to specify that `"data"` is an optional field we do want to `pre_initialize`.